File: task-03/Prime.rb
num = gets.to_i
 
 list=[2]
 
 for i in 3..num do
     a=0
     list.each do |j|
         if i%j!=0
             a+=1
         end
     end
    if a==list.length()
        list.append(i)
    end
        
 end
 
if num>=2
    puts list
end

File: task-03/prime.exs
defmodule PrimeArray do
  def get_primes do
    IO.puts("Enter a number:")
    input = IO.gets("") |> String.trim() |> String.to_integer()
    
    case input do
      n when n < 2 ->
        IO.puts("Please enter a valid number greater than or equal to 2.")
        get_primes()
      n ->
        primes = Enum.filter(2..n, &is_prime(&1))
        IO.puts("#{Enum.join(primes, ", ")}")
    end
  end

  defp is_prime(n) when n in [2, 3], do: true
  defp is_prime(n) do
    floored_sqrt = :math.sqrt(n)
      |> Float.floor
      |> round
    !Enum.any?(2..floored_sqrt, &(rem(n, &1) == 0))
  end
end


PrimeArray.get_primes()
